packages/contracts/contracts/L1/AssetManager.sol

```diff
@@ -884,24 +725,13 @@ contract AssetManager is ISemver, IERC721Receiver, IAssetManager {
      * @param validator Address of the validator.
      * @param delegator Address of the delegator.
      * @param kghCount  The number of KGHs to delegate.
-     * @param kroInKghs The amount of KRO in the KGHs.
-     * @param kroShares The amount of KRO shares to receive.
      */
-    function _delegateKghBatch(
-        address validator,
-        address delegator,
-        uint128 kghCount,
-        uint128 kroInKghs,
-        uint128 kroShares
-    ) internal {
-        Asset storage asset = _vaults[validator].asset;
-        KghDelegator storage kghDelegator = _vaults[validator].kghDelegators[delegator];
+    function _delegateKghBatch(address validator, address delegator, uint128 kghCount) internal {
+        Vault storage vault = _vaults[validator];
+        KghDelegator storage kghDelegator = vault.kghDelegators[delegator];
```

Comment:
How about removing this and just use `vault.kghDelegators[delegator].kghNum` at L736?